xbmc/guilib/Texture.cpp

```diff
@@ -68,21 +68,12 @@ void CBaseTexture::Allocate(unsigned int width, unsigned int height, unsigned in
   m_textureWidth = m_imageWidth;
   m_textureHeight = m_imageHeight;
 
-  if (m_format & XB_FMT_DXT_MASK)
-    while (GetPitch() < CServiceBroker::GetRenderSystem().GetMinDXTPitch())
-      m_textureWidth += GetBlockSize();
-
-  if (!CServiceBroker::GetRenderSystem().SupportsNPOT((m_format & XB_FMT_DXT_MASK) != 0))
+  if (!CServiceBroker::GetRenderSystem().SupportsNPOT())
   {
     m_textureWidth = PadPow2(m_textureWidth);
     m_textureHeight = PadPow2(m_textureHeight);
   }
-  if (m_format & XB_FMT_DXT_MASK)
```

Comment:
I'm not sure that this will not break windows at least on older hw.

Reply:
If windows requires this, we should hide it in platform code. What do you think?

Reply:
I was wrong when saying that this breaks windows. look like this breaks support of compressed formats (XB_FMT_DXT(1|2|5)) at all, because it's a requirement for texture for these formats.
I'm not sure for OGL (I think it acts the same) but MSDN says:
> Direct3D will not allow a resource to be created using BC format with the top-level size set to something other than a multiple of 4 in width and height, even though it does allow the mipchain below it to not meet that requirement. 

if we drop this we also must drop switch at https://github.com/xbmc/xbmc/pull/13512/files#diff-a438d95a90f1e375829fdccadc6f1e1cL122
I'm also not sure that this will work if it will try to create a texture with `GL_COMPRESSED_RGBA_S3TC_DXT1_EXT` format via `glTexImage2D` instead of `glCompressedTexImage2D`

Reply:
I'm not against removing render caps, I want to do this right. I think removing XB_FMT_DXT_MASK is wrong. because we do support of DDS images which can be in compressed format.

Reply:
I missed this as I searched the code for this. I'll put it back